src/GraphManager/GraphEdit/ButtonTests.test.tsx

```diff
@@ -0,0 +1,149 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { NoTouchButton } from "./NoTouchButton";
+import { UserSettingsButton } from "./UserSettingsButton";
+import { EditModeButton } from "./ModeButton";
+import { CreateButton} from "./CreateButton";
+import { Controller } from "./GraphEdit";
+
+// Mock functions from GraphEdit
+jest.mock("./GraphEdit", () => ({
+  openCreateNodePopUpAtPagePosition: jest.fn(),
+  openCreateLinkPopUp: jest.fn(),
+  i18n: { t: jest.fn((key: string) => key) }, // Mock translation function
+}));
+
+// Mock User Data Context
+jest.mock("@src/Context/UserDataContext", () => ({
+  useUserDataContext: jest.fn(),
+}));
+
+import { useUserDataContext } from "@src/Context/UserDataContext";
+
+describe("Button Components", () => {
+  let mockController: Controller;
+  let setIsOpen: jest.Mock;
+  let mockDisplayAlert: jest.Mock;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    // Configure the mocked useUserDataContext
+    (useUserDataContext as jest.Mock).mockReturnValue({ userID: null });
+
+    // Mock necessary parts of the Controller
+    mockController = {
+      popUp: { state: { isOpen: false }, setState: jest.fn() },
+      graph: {
+        current: {
+          nodes: [],
+          edges: [],
+        },
+      },
+      language: "en",
+      mode: {
+        isEditingEnabled: true,
+        setIsEditingEnabled: jest.fn(),
+        allowGraphInteractions: true,
+        setAllowGraphInteractions: jest.fn(),
+        use3D: false,
+        setUse3D: jest.fn(),
+      },
+      search: {
+        isResultShown: false,
+        setIsResultShown: jest.fn(),
+        highlightNodes: new Set(),
+        setHighlightNodes: jest.fn(),
+      },
+    } as unknown as Controller;
+
+    setIsOpen = jest.fn();
+    mockDisplayAlert = jest.fn();
+    global.alert = mockDisplayAlert; // Mock global alert
+  });
+
+  it("should toggle allowGraphInteractions when NoTouchButton is clicked", async () => {
+    render(<NoTouchButton ctrl={mockController} />);
+    const user = userEvent.setup();
+
+    const button = screen.getByRole("button", { name: /View-only mode|Enable Graph-Interaction/i });
+    await user.click(button);
+
+    expect(mockController.mode.setAllowGraphInteractions).toHaveBeenCalledWith(
+      expect.any(Function)
```

Comment:
shouldn't we expect the argument to be always`true`? otherwise there is a function `toHaveBeenCalled()` that doesn't care about the arguments given, that might be more readable :)